core/sql/optimizer/NATable.cpp

```diff
@@ -86,6 +86,10 @@
 #include "CmpMain.h"
 
 #define MAX_NODE_NAME 9
+#define MAX_PRECISION_ALLOWED  18
+#define HIVE_MAX_PRECISION_ALLOWED  38
+#define MAX_SCALE_ALLOWED     6
```

Comment:
I don't think there is a maximum scale, neither for Hive nor for Trafodion. The only condition right now is that the scale can't exceed the precision. Example of a valid scale: DECIMAL(18,18).